source/postprocess/sealevel.cc

```diff
@@ -0,0 +1,655 @@
+                  }
+              }
+          }
+      
+      const double sealevel_offset = -1./integral_oceanmask*(1./density_water*integral_iceheight+integral_topo_geoid);
+
+      std::vector<std::vector<double>> topo_spherical_function;
+      std::vector<std::vector<double>> geoid_spherical_function;
+      std::vector<std::vector<double>> oceanmask_spherical_function;
+      std::vector<std::vector<double>> iceheight_spherical_function;
+
+      for (unsigned int i=0; i<topo_stored_values.size(); ++i)
+        {
+          const std::array<double,3> scoord = aspect::Utilities::Coordinates::cartesian_to_spherical_coordinates(topo_stored_values[i].first);
+
+          // Calculate spherical infinitesimal sin(theta)*d_theta*d_phi by infinitesimal_area/radius^2
+          const double infinitesimal = topo_stored_values[i].second.first/(outer_radius*outer_radius);
+
+          // Theta, phi, spherical infinitesimal, and topography
+          topo_spherical_function.emplace_back(std::vector<double> {scoord[2],
+                                                                    scoord[1],
+                                                                    infinitesimal,
+                                                                    topo_stored_values[i].second.second
+                                                                   });
+          
+          // Theta, phi, spherical infinitesimal, and geoid displacement  
+          geoid_spherical_function.emplace_back(std::vector<double> {scoord[2],
+                                                                     scoord[1],
+                                                                     infinitesimal,
+                                                                     geoid_stored_values[i].second.second
+                                                                    });
+
+          // Theta, phi, spherical infinitesimal, and ocean mask  
+          oceanmask_spherical_function.emplace_back(std::vector<double> {scoord[2],
+                                                                         scoord[1],
+                                                                         infinitesimal,
+                                                                         oceanmask_stored_values[i].second.second
+                                                                        });
+
+          // Theta, phi, spherical infinitesimal, and ice height
+          iceheight_spherical_function.emplace_back(std::vector<double> {scoord[2],
+                                                                         scoord[1],
+                                                                         infinitesimal,
+                                                                         iceheight_stored_values[i].second.second
+                                                                        });
+        }
```

Comment:
It looks like you are not using these lines. Maybe remove? Or where do you plan to use them?

Reply:
You're right! I do want to write these variables, but I'll figure that out in the execute function